src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -243,9 +243,19 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
   }
 
   override def loadSubjectEmail(subject: WorkbenchSubject): Future[Option[WorkbenchEmail]] = withContext { ctx =>
-    val subDn = subjectDn(subject)
-    Option(ctx.getAttributes(subDn).get(Attr.email)).map { emailAttr =>
-      WorkbenchEmail(emailAttr.get.asInstanceOf[String])
+    Try {
+      val subDn = subjectDn(subject)
+      Option(ctx.getAttributes(subDn).get(Attr.email)).map { emailAttr =>
+        WorkbenchEmail(emailAttr.get.asInstanceOf[String])
+      }
+    }.recover {
+      case _: NameNotFoundException => None
+    }.get
+  }
+
+  override def loadSubjectEmails(subjects: Set[WorkbenchSubject]): Future[Set[WorkbenchEmail]] = {
+    Future.traverse(subjects) { subject =>
```

Comment:
look at loadGroups for how to do batched queries... this scales with the number of subjects which probably won't usually be that bad but there will be cases where it is.

Reply:
also throwing an exception when a subject isn't found means that if somehow a group has a reference to a subject that doesn't exist this call will perpetually break. Maybe return a Map[Subject, Option[Email]) and let the caller figure it out? Or just ignore (which is what happens in the batch load).

Reply:
In order to do this all in one query, I think we would need to search the `groupsOu` and `peopleOu` namespaces in a single query and I'm not sure how to do that.  Instead, I simply proxy out the queries to the existing `loadUsers` and `loadGroups` methods.  This is an improvement because we're down to 2 queries instead of 1 per subject, which is better, but admittedly we might be able to do better.  If there's something you can steer me towards to get this done in a single query, please just point me in the right direction.